File: PlayTogether/PlayTogether.Domain/ContactRequest.cs
using System;
using System.ComponentModel.DataAnnotations.Schema;

namespace PlayTogether.Domain
{
    public class ContactRequest: ISimpleEntity
    {
        [DatabaseGenerated(DatabaseGeneratedOption.Identity)]
        public Guid Id { get; set; }
        public ContactRequestStatus Status { get; set; }
        public Guid UserId { get; set; }
        public Guid ToUserId { get; set; }

        public virtual User User { get; set; }
    }
}

File: PlayTogether/PlayTogether.Web/ClientApp/src/app/models/contact-request.ts
export class ContactRequest {
  id: string;
  toUserId: string;
  userId: string;
  fromUserName: string;
  toUserName: string;
  status: ContactRequestStatus;
  isUserOwner: boolean;
}

export enum ContactRequestStatus {
  Open = 0,
  Approved,
  Rejected
}
